platform/src/main/java/org/stellar/anchor/platform/custody/service/FireblocksEventsService.java

```diff
@@ -0,0 +1,94 @@
+package org.stellar.anchor.platform.custody.service;
+
+import static org.stellar.anchor.platform.utils.RSAUtil.RSA_ALGORITHM;
+import static org.stellar.anchor.platform.utils.RSAUtil.SHA512_WITH_RSA_ALGORITHM;
+import static org.stellar.anchor.platform.utils.RSAUtil.generatePublicKey;
+import static org.stellar.anchor.platform.utils.RSAUtil.isValidSignature;
+import static org.stellar.anchor.util.Log.debugF;
+import static org.stellar.anchor.util.Log.info;
+import static org.stellar.anchor.util.StringHelper.isEmpty;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import org.stellar.anchor.api.custody.fireblocks.FireblocksEventObject;
+import org.stellar.anchor.api.exception.BadRequestException;
+import org.stellar.anchor.api.exception.InvalidConfigException;
+import org.stellar.anchor.platform.config.FireblocksConfig;
+import org.stellar.anchor.util.GsonUtils;
+
+@AllArgsConstructor
+public class FireblocksEventsService {
+
+  public static final String FIREBLOCKS_SIGNATURE_HEADER = "fireblocks-signature";
+
+  private final PublicKey publicKey;
+
+  public FireblocksEventsService(FireblocksConfig fireblocksConfig) throws InvalidConfigException {
+    publicKey = getFireblocksPublicKey(fireblocksConfig.getPublicKey());
```

Comment:
What do you think about moving getFireblocksPublicKey into FireblocksConfig and doing validation there?